build_library/build_image_util.sh

```diff
@@ -58,10 +58,28 @@ extract_update() {
   local image_name="$1"
   local disk_layout="$2"
   local update_path="${BUILD_DIR}/${image_name%_image.bin}_update.bin"
+  local digest_path="${update_path}.DIGESTS"
 
   "${BUILD_LIBRARY_DIR}/disk_util" --disk_layout="${disk_layout}" \
     extract "${BUILD_DIR}/${image_name}" "USR-A" "${update_path}"
-  upload_image "${update_path}"
+
+  # Compress image
+  image_files+=( "${update_path}" )
+  declare -a compressed_images
+  compress_disk_images image_files compressed_images
+
+  # Upload compressed image
+  upload_image -d "${digest_path}" ${compressed_images[@]}
```

Comment:
What if the image wasn't compressed? This should upload the `image_files` entry then. I suggest you add an "else" case to `compress_disk_images` where the check for the img/vmdk etc extension is done and add the plain file otherwise. Now that means the naming of "compressed_images" is a bit wrong and we can rather call it "images_to_upload".

Reply:
Ok, then I suggest to have two outputs, one is compressed, one is the list of skipped files and then the `upload_image` command can pass the concatenation of both and this also helps to remove the duplication in `vm_image_util.sh`, or?

Reply:
let's give that a shot.

Reply:
updated